applications/TopologyOptimizationApplication/CMakeLists.txt

```diff
@@ -0,0 +1,80 @@
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+message("**** configuring KratosTopologyOptimizationApplication ****")
+
+################### PYBIND11
+include(pybind11Tools)
+
+kratos_add_dependency(${KRATOS_SOURCE_DIR}/applications/StructuralMechanicsApplication)
+
+include_directories( ${KRATOS_SOURCE_DIR}/kratos )
+include_directories( ${KRATOS_SOURCE_DIR}/applications/StructuralMechanicsApplication )
+include_directories( ${KRATOS_SOURCE_DIR}/applications/TopologyOptimizationApplication )
+
+## TopologyOptimization core sources
+set( KRATOS_TOPOLOGY_OPTIMIZATION_APPLICATION_CORE     
+	${CMAKE_CURRENT_SOURCE_DIR}/topology_optimization_application.cpp 
+    	${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/small_displacement_simp_element.cpp
+	)
+
+## TopoloyOptimization python interface sources
+set( KRATOS_TOPOLOGY_OPTIMIZATION_APPLICATION_PYTHON_INTERFACE	
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/topology_optimization_python_application.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_utilities_to_python.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_strategies_to_python.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/small_displacement_simp_element.cpp
+)
+
+add_library(KratosTopologyOptimizationCore SHARED ${KRATOS_TOPOLOGY_OPTIMIZATION_APPLICATION_CORE})
+target_link_libraries(KratosTopologyOptimizationCore PRIVATE KratosCore KratosStructuralMechanicsCore)
+set_target_properties(KratosTopologyOptimizationCore PROPERTIES COMPILE_DEFINITIONS "TOPOLOGY_OPTIMIZATION_APPLICATION=EXPORT,API")
+
+###############################################################
+## define library Kratos which defines the basic python interface
+pybind11_add_module(KratosTopologyOptimizationApplication MODULE THIN_LTO ${KRATOS_TOPOLOGY_OPTIMIZATION_APPLICATION_PYTHON_INTERFACE})
+target_link_libraries(KratosTopologyOptimizationApplication PRIVATE KratosTopologyOptimizationCore)
+set_target_properties(KratosTopologyOptimizationApplication PROPERTIES PREFIX "")
+
+# changing the .dll suffix to .pyd 
+if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+    set_target_properties(KratosTopologyOptimizationApplication PROPERTIES SUFFIX .pyd)
+endif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+
+# changing the .dylib suffix to .so
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set_target_properties(KratosTopologyOptimizationApplication PROPERTIES SUFFIX .so)
+endif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+
+if(${INSTALL_PYTHON_FILES} MATCHES ON)
+  get_filename_component (CURRENT_DIR_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/python_scripts DESTINATION applications/${CURRENT_DIR_NAME}  FILES_MATCHING PATTERN "*.py"  PATTERN ".svn" EXCLUDE)
+endif(${INSTALL_PYTHON_FILES} MATCHES ON)
+
+# Cotire
+if(USE_COTIRE MATCHES ON)
+    cotire(KratosTopologyOptimizationCore)
+    cotire(KratosTopologyOptimizationApplication)
+endif(USE_COTIRE MATCHES ON)
```

Comment:
cotire is not used anymore

Reply:
I suggest to compare the file with the initial commit version (thats what was generated from the appgenerator) and reduce differences if possible.

Reply:
https://github.com/KratosMultiphysics/Kratos/pull/9480/files/c600c35b43773a7f0f85af7e4f29e96543bf33d4..1114f4ed52dd1928130e584ffa5136c4fbf8cc98